### ISSUE TITLE
Content don't refresh when composants are in a slot

### ISSUE THREAD

**Issue by Barbapapazes** [nuxt/content#1831]
### Environment

Nuxi 3.0.0

RootDir: /home/projects/github-vanssr
Nuxt project info: 

------------------------------
- Operating System: `Linux`
- Node Version:     `v16.14.2`
- Nuxt Version:     `3.0.0`
- Nitro Version:    `1.0.0`
- Package Manager:  `npm@7.17.0`
- Builder:          `vite`
- User Config:      `modules`, `content`
- Runtime Modules:  `@nuxt/content@2.3.0`
- Build Modules:    `-`
------------------------------

### Reproduction

https://stackblitz.com/edit/github-vanssr?file=content/index.md

### Describe the bug

When you encapsulate a ContentSlot inside another component (which use a slot), the preview will not be update in dev mode.

### Additional context

Feel free to play with the simple reproduction to understand. I struggle to explain the issue.

**Comment by nobkd** [comment]
If you insert `{{ $slots.title }}` in your component, you see this for a split second (on page reloads):

```ts
 (...args) => {
        // If a user calls a compiled slot inside a template expression (#1745), it
        // can mess up block tracking, so by default we disable block tracking and
        // force bail out when invoking a compiled slot (indicated by the ._d flag).
        // This isn't necessary if rendering a compiled `<slot>`, so we flip the
        // ._d flag off when invoking the wrapped fn inside `renderSlot`.
        if (renderFnWithContext._d) {
            setBlockTracking(-1);
        }
        const prevInstance = setCurrentRenderingInstance(ctx);
        let res;
        try {
            res = fn(...args);
        }
        finally {
            setCurrentRenderingInstance(prevInstance);
            if (renderFnWithContext._d) {
                setBlockTracking(1);
            }
        }
        {
            devtoolsComponentUpdated(ctx);
        }
        return res;
    } 
```

Also, if you try to display the slots with `{{ $slots }}`, the object is empty, which might explain your problem (with help of the log), as they are computed the first time but then ignored, I think?

---

It seems like the update failure is connected to Vue:
Here's the issue mentioned in the code: https://github.com/vuejs/core/issues/1745

I have no idea how this could be fixed for dev mode.

**Comment by Barbapapazes** [comment]
Hello,

After more research, it seems that it's not possible to have a component inside a component used in a markdown file. The strange part, is that this pattern is used in the nuxt website, https://www.npmjs.com/package/nuxt-website-theme?activeTab=explore, but using an external lib. So maybe, it's the missing part?

**Comment by Barbapapazes** [comment]
Okkk, very strange. If you add comments around your component, it will work.


This works
```vue
<template>
  <!-- <div> -->
    <BaseSection>
      <ContentSlot :use="$slots.default" unwrap="p" />
    </BaseSection>
  <!-- </div> -->
</template>
```

But this not
```vue
<template>
  <div>
    <BaseSection>
      <ContentSlot :use="$slots.default" unwrap="p" />
    </BaseSection>
  </div>
</template>
```

**Comment by farnabaz** [comment]
This issue is fixed in `^2.8`.
Checkout: https://stackblitz.com/edit/github-vanssr-eke8fm?file=content%2Findex.md

**Comment by Barbapapazes** [comment]
Holy sh*t, that is so cool! 🙌

**Comment by Barbapapazes** [comment]
> This issue is fixed in `^2.8`.
> Checkout: https://stackblitz.com/edit/github-vanssr-eke8fm?file=content%2Findex.md

Holy sh*t, that is so cool! 🙌
